app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassKotlinActivity.kt

```diff
@@ -0,0 +1,119 @@
+import com.getsensibill.web.data.UiFinishReason
+import com.getsensibill.web.data.configuration.NavigationIntent
+import com.getsensibill.web.data.configuration.ProgrammaticTheme
+import com.getsensibill.web.data.models.Brand
+import com.getsensibill.web.ui.WebUiActivity
+import com.getsensibill.web.ui.WebUiFragment
+import com.getsensibill.web.ui.WebUiNetworkErrorFragment
+import timber.log.Timber
+
+/**
+ * Subclassing the WebUiActivity
+ * <p>
+ * When you subclass the WebUiActivity, you do not need to override anything for it to work.
+ * It will use all the defaults provided by the [WebUiActivity] and the [WebUiFragment].
+ *
+ * If you are using your own view, then its important to override the [callSetContentView] and
+ * the [webContainerId] so that the [WebUiActivity] knows where to place the [WebUiFragment].
+ */
+class SubclassKotlinActivity : WebUiActivity(), WebUiFragment.Listener {
+    private lateinit var binding: ActivitySubclassKotlinBinding
+
+    /**
+     * Not required override.
+     *
+     * Can be used to override the setContentView, so that it uses your provided
+     * activities view, instead of the default [WebUiActivity] view.
+     */
+    override fun callSetContentView() {
+        binding = ActivitySubclassKotlinBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+    }
+
+    /**
+     * Not required override, unless you are overriding [callSetContentView]
+     *
+     * If we are overriding the setContentView, we also need to set the container id
+     * which will contain the [WebUiFragment].
+     */
+    override val webContainerId: Int
+        get() = binding.webContainer.id
+
+    /**
+     * Not required override.
+     *
+     * If you want to change the default navigation when first loading into Spend Manager,
+     * you can override this and provide a different [NavigationIntent]. Alternatively,
+     * you can also apply this as an argument when overriding [createWebUiFragment].
+     */
+    override val webInitialNavigation: NavigationIntent
+        get() = NavigationIntent.DASHBOARD
+
+    /**
+     * Not required override.
+     *
+     * Used to create the [WebUiFragment] held in the Activity. Common reason for overriding
+     * this would be if there are arguments that are needed to be supplied to the fragment.
+     */
+    override fun createWebUiFragment(): WebUiFragment = WebUiFragment().apply {
+        arguments = Bundle().apply {
+            // Pass in a navigation override. Defaults as .DASHBOARD
+            putParcelable(WebUiFragment.ARG_NAVIGATION_OVERRIDE, NavigationIntent.DASHBOARD)
+            // Pass in a custom Theme override. Defaults to null
+            putParcelable(WebUiFragment.ARG_PROGRAMMATIC_THEME_OVERRIDE, ProgrammaticTheme(Brand()))
+        }
+    }
+
+    /**
+     * Not required override.
+     *
+     * The shouldNavigateBack from the [webFragment] will inform you if your activity
```

Comment:
In the comments here we should probably mention `activityOnBackPressed` in case the client wants default back functionality

Reply:
updated